### ISSUE TITLE
304 Not Modified response includes body

### ISSUE THREAD

**Issue by phooijenga** [Unleash/unleash-edge#341]
### Describe the bug

After adding Unleash to one of our Go projects we noticed a stream of log messages like these:

```
2023/11/22 12:07:53 Unsolicited response received on idle HTTP channel starting with "14\r\n\x1f\x8b\b\x00\x00\x00\x00\x00\x04\xff\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n0\r\n\r\n"; err=<nil>
2023/11/22 12:08:08 Unsolicited response received on idle HTTP channel starting with "14\r\n\x1f\x8b\b\x00\x00\x00\x00\x00\x04\xff\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n0\r\n\r\n"; err=<nil>
2023/11/22 12:08:23 Unsolicited response received on idle HTTP channel starting with "14\r\n\x1f\x8b\b\x00\x00\x00\x00\x00\x04\xff\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n0\r\n\r\n"; err=<nil>
2023/11/22 12:08:38 Unsolicited response received on idle HTTP channel starting with "14\r\n\x1f\x8b\b\x00\x00\x00\x00\x00\x04\xff\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n0\r\n\r\n"; err=<nil>
2023/11/22 12:08:53 Unsolicited response received on idle HTTP channel starting with "14\r\n\x1f\x8b\b\x00\x00\x00\x00\x00\x04\xff\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n0\r\n\r\n"; err=<nil>
2023/11/22 12:09:08 Unsolicited response received on idle HTTP channel starting with "14\r\n\x1f\x8b\b\x00\x00\x00\x00\x00\x04\xff\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n0\r\n\r\n"; err=<nil>
```

After some further investigation we found that the 304 Not Modified responses from Unleash Edge include a chunked and compressed response body (the 14... and further). 

[RFC 9110 § 15.4.5](https://www.rfc-editor.org/rfc/rfc9110#status.304) says:

> A 304 response is terminated by the end of the header section; it cannot contain content or trailers.

### Steps to reproduce the bug

1. Connect to Unleash Edge using [unleash-client-go](https://github.com/Unleash/unleash-client-go)

### Screenshots

![image](https://github.com/Unleash/unleash-edge/assets/135009186/4d10e8a3-daf0-490e-acdc-df80d74fbf78)

### Unleash version

v5.6.7, Edge v16.0.3, Go Client v4.3.1

### Subscription type

None

### Hosting type

None

**Comment by phooijenga** [comment]
It looks like this is caused by the Compress middleware compressing the empty response generated by the Etag middleware.

**Comment by kwasniew** [comment]
Thanks for the report. It is spot on and we were able to reproduce it with Go SDK. You're right that the order of middleware fns matters. We can't just swap the order of Etag and Compress since it introduces another bug. 
We'll start looking into possible solutions soon.

**Comment by chriswk** [comment]
Since @phooijenga was so kind as to make a PR with the fix against our ETag middleware, I've made a new release with that fix, and a PR is in the works #342
